internal/grpc/services/gateway/storageprovidercache.go

```diff
@@ -22,30 +22,70 @@ import (
 	"context"
 	"encoding/json"
 	"strings"
+	"time"
 
 	"github.com/ReneKroon/ttlcache/v2"
 	userpb "github.com/cs3org/go-cs3apis/cs3/identity/user/v1beta1"
 	rpc "github.com/cs3org/go-cs3apis/cs3/rpc/v1beta1"
 	provider "github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1"
+	registry "github.com/cs3org/go-cs3apis/cs3/storage/registry/v1beta1"
 	ctxpkg "github.com/cs3org/reva/pkg/ctx"
+	sdk "github.com/cs3org/reva/pkg/sdk/common"
 	"github.com/cs3org/reva/pkg/utils"
 	"google.golang.org/grpc"
 )
 
 // generates a user specific key pointing to ref
-func userKey(ctx context.Context, ref *provider.Reference) string {
+func userKey(user *userpb.User, ref *provider.Reference) string {
 	if ref == nil || ref.ResourceId == nil || ref.ResourceId.StorageId == "" {
 		return ""
 	}
-	u, ok := ctxpkg.ContextGetUser(ctx)
-	if !ok {
-		return ""
+
+	return "uid" + user.Id.OpaqueId + "!sid:" + ref.ResourceId.StorageId + "!oid:" + ref.ResourceId.OpaqueId + "!path:" + ref.Path
+}
+
+// Caches holds all caches used by the gateway
+type Caches struct {
```

Comment:
I have a conflict with this type. Could we define `Caches` as: `type Caches [3]*ttlcache.Cache`? or an enum, in case accessing by index loses meaning, and accessing it can be done like:


```go
const (
	stat = iota
	home
	provider
)

type Caches [3]*ttlcache.Cache

// only access
func main() {
	c := Caches{}
	statCache := c[stat]
	homeCache := c[home]
	providerCache := c[provider]
}
```